migrations/versions/41b3af7e4449_migrating_preview_video_to_video_field.py

```diff
@@ -0,0 +1,129 @@
+            if urls:
+                if len(urls) > 1:
+                    troublesome_previews.append(
+                        {
+                            'proposal_id': prop['id'],
+                            'preview_video': prop['preview_video'],
+                        }
+                    )
+                try:
+                    video_source, video_id = parse_video_url(urls[0])
+                    conn.execute(
+                        sa.update(proposal)
+                        .where(proposal.c.id == prop['id'])
+                        .values(video_source=video_source, video_id=video_id)
+                    )
+                except ValueError:
+                    troublesome_previews.append(
+                        {
+                            'proposal_id': prop['id'],
+                            'preview_video': prop['preview_video'],
+                        }
+                    )
+
+    with open(troublesome_filename, 'w') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=['proposal_id', 'preview_video'])
+        writer.writeheader()
+        writer.writerows(troublesome_previews)
+
+    op.execute(
+        sa.DDL(
+            dedent(
+                '''
+        UPDATE proposal SET search_vector = setweight(to_tsvector('english', COALESCE(title, '')), 'A') || setweight(to_tsvector('english', COALESCE(abstract_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(outline_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(requirements_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(slides, '')), 'B') || setweight(to_tsvector('english', COALESCE(links, '')), 'B') || setweight(to_tsvector('english', COALESCE(bio_text, '')), 'B');
+
+        DROP TRIGGER proposal_search_vector_trigger ON proposal;
+        DROP FUNCTION proposal_search_vector_update();
+
+        CREATE FUNCTION proposal_search_vector_update() RETURNS trigger AS $$
+        BEGIN
+            NEW.search_vector := setweight(to_tsvector('english', COALESCE(NEW.title, '')), 'A') || setweight(to_tsvector('english', COALESCE(NEW.abstract_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(NEW.outline_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(NEW.requirements_text, '')), 'B') || setweight(to_tsvector('english', COALESCE(NEW.slides, '')), 'B') || setweight(to_tsvector('english', COALESCE(NEW.links, '')), 'B') || setweight(to_tsvector('english', COALESCE(NEW.bio_text, '')), 'B');
+            RETURN NEW;
+        END
+        $$ LANGUAGE plpgsql;
+
+        CREATE TRIGGER proposal_search_vector_trigger BEFORE INSERT OR UPDATE ON proposal
+        FOR EACH ROW EXECUTE PROCEDURE proposal_search_vector_update();
+                '''
+            )
+        )
+    )
+
+    op.drop_column('proposal', 'preview_video')
+
+
+def downgrade():
+    op.add_column(
+        'proposal', sa.Column('preview_video', sa.UnicodeText(), nullable=True)
+    )
```

Comment:
This needs to be reverse-populated from the video_id/video_source columns. Also, where is the migration to add them?

Reply:
Added that